Lib/test/test_contextlib.py

```diff
@@ -306,6 +306,88 @@ def woohoo():
             with woohoo():
                 raise StopIteration
 
+    def test_contextmanager_handling_exception_resets_exc_info(self):
+        # Test that sys.exc_info() is correctly unset after handling the error
+        # when used within a context manager
+
+        @contextmanager
+        def ctx(reraise=False):
+            try:
+                self.assertIsNone(sys.exception())
+                yield
+            except:
+                self.assertIsInstance(sys.exception(), ZeroDivisionError)
+                if reraise:
+                    raise
+            else:
+                self.assertIsNone(sys.exception())
+            self.assertIsNone(sys.exception())
+
+        with ctx():
+            pass
+
+        with ctx():
+            1/0
+
+        with self.assertRaises(ZeroDivisionError):
+            with ctx(reraise=True):
+                1/0
```

Comment:
I think we need to check here that the context exception of the ZeroDivisionError is correct.

Reply:
Sorry, I'm not sure exactly what you mean. Do you mean the `__context__` attribute of the error should be checked to make sure it's `None`? Or that after the `with` block exits that `sys.exception()` is `None`?